### PR TITLE
feat: new touchMode options

### LINKED ISSUE #3: Style does not change in watch mode
If I change the style during development (watch), it won't be adopted.

Here is a repo to reproduce.: https://github.com/deleonio/stencil-meets-windicss

### PR DESCRIPTION
**Pull request by deleonio**
Fix: https://github.com/windicss/postcss-windicss/issues/3

### CODE REVIEW

File: src/dev.ts [hunk 1]
```diff
@@ -38,7 +46,7 @@ export async function startDevWatcher() {
       }
 
       debug('update from', path)
-      await utils!.extractFile(await fs.readFile(path, 'utf-8'))
+      await utils!.extractFile(fs.readFileSync(path, 'utf-8'))
```
Comment (antfu) [inline src/dev.ts]: Why do we need to change this to sync?

**Review by antfu — CHANGES_REQUESTED** [review]
I was thinking the same approach, thanks for bringing this up.

But since the mtime touch works for most of the dev tools and changing the source file will introduce unwanted git diffs. I think this behavior is better under and options (default to `false`) where you can enable explicitly for the tools that check the content of the source file on HMR.

And also we can reuse the `touch` function in utils by passing a second arg show if we should append the timestamp comment.

File: src/dev.ts (hunk 1 above)
Comment (deleonio) [reply src/dev.ts]: I chose the sync variant because it seems easier to me and we don't wait for anything.

Comment (antfu) [reply src/dev.ts]: `readFileSync` blocks the entire process when waiting for the fs, it's actually slowing down the entire process. While this is not significant, I don't think we need to change this line if the existing code works fine and there is no improvement. Can you revert this?

Comment (deleonio) [reply src/dev.ts]: jep.

Comment (deleonio) [reply src/dev.ts]: @antfu Is the async handling better now?

Comment (deleonio) [reply src/dev.ts]: - [ ] I think I have to consolidate the touch function.

**Comment by deleonio** [comment]
I think it should fit like this?!

**Comment by antfu** [comment]
Made some changes but overall it's great. Thanks you!

**Pull request merged**